docs/guides/features/ipsec-offload.md

```diff
@@ -0,0 +1,138 @@
+# IPsec Offload
+
+The Inline IPsec offload feature uses the
+[IPDK Infrastructure Application Interface](https://ipdk.io/documentation/Interfaces/InfraApp/)
+to enable cryptographically-secure data traffic. The IPsec control plane
+(IKE protocol) is offloaded, thus avoiding specialized drivers or control
+planes burdening compute instances. Further details are available on the
+[IPDK Inline Acceleration - IPsec](https://ipdk.io/documentation/Recipes/InlineIPsec/)
+page.
+
+This feature works in conjunction with the
+[IPsec Recipe](https://github.com/ipdk-io/ipsec-recipe), which provides
+strongSwan plugin code to configure and program control messages into the
+target.
+
+The IPsec offload feature is only supported on the Intel&reg; IPU E2100 target.
+
+## Feature overview
+
+The `infrap4d` process provides the gRPC server-side support for P4RT and
+gNMI messages. strongSwan acts as the gRPC client in this context.
+You can also use other clients that implement the IKE stack. The
+client uses P4RT to program the Security Policy Database (SPD), and it
+uses gNMI to configure Security Association Database (SAD) entries, which
+includes encryption keys, re-keying etc.
+
+The [openconfig-ipsec-offload](https://github.com/ipdk-io/openconfig-public/blob/master/release/models/ipsec/openconfig-ipsec-offload.yang)
+YANG model is used to configure IPsec offload.
+
+## Enabling IPsec
+
+Follow the sequence of steps listed below to enable IPsec functionality.
+
+### Load crypto package
```

Comment:
Load IPSEC P4 package?

Also, should we say something about using IPU SDK and IPSEC P4 program and follow compiling-p4-programs.md to obtain this IPSEC P4 package?

Reply:
I have moved the Compile P4 before this, since that is needed to get the pkg file. I am leveraging the other documents (Compiling P4 programs, Deploying P4 programs, and Running infrap4d documents) in order to avoid repeating text.